client/my-sites/plugins/single-plugin.jsx

```diff
@@ -0,0 +1,374 @@
+import PluginSiteList from 'calypso/my-sites/plugins/plugin-site-list';
+import { siteObjectsToSiteIds } from 'calypso/my-sites/plugins/utils';
+import SidebarNavigation from 'calypso/my-sites/sidebar-navigation';
+import { recordGoogleEvent } from 'calypso/state/analytics/actions';
+import getToursHistory from 'calypso/state/guided-tours/selectors/get-tours-history';
+import {
+	getPluginOnSite,
+	getPluginOnSites,
+	getSiteObjectsWithPlugin,
+	getSitesWithoutPlugin,
+	isPluginActionInProgress,
+	isRequestingForSites,
+} from 'calypso/state/plugins/installed/selectors';
+import { fetchPluginData as wporgFetchPluginData } from 'calypso/state/plugins/wporg/actions';
+import {
+	isFetching as isWporgPluginFetching,
+	isFetched as isWporgPluginFetched,
+	getPlugin as getWporgPlugin,
+} from 'calypso/state/plugins/wporg/selectors';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import canCurrentUserManagePlugins from 'calypso/state/selectors/can-current-user-manage-plugins';
+import getSelectedOrAllSitesWithPlugins from 'calypso/state/selectors/get-selected-or-all-sites-with-plugins';
+import hasNavigated from 'calypso/state/selectors/has-navigated';
+import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
+import { isJetpackSite, isRequestingSites } from 'calypso/state/sites/selectors';
+import { getSelectedSite, getSelectedSiteId } from 'calypso/state/ui/selectors';
+import NoPermissionsError from './no-permissions-error';
+import { useLocalizedMoment } from 'calypso/components/localized-moment';
+import { Button } from '@automattic/components';
+
+function SinglePlugin( props ) {
+	const {
+		wporgFetched: isFetched,
+		wporgFetching: isFetching,
+		plugin,
+		wporgPlugin,
+		selectedSite,
+		siteIds,
+		translate,
+		sitesWithPlugin,
+		requestingPluginsForSites,
+	} = props;
+
+	const moment = useLocalizedMoment();
+
+	useEffect( () => {
+		if ( ! isFetched ) {
+			props.wporgFetchPluginData( props.pluginSlug );
+		}
+	}, [ isFetched ] );
+
+	const fullPlugin = useMemo( () => {
+		// assign it .org details
+		return {
+			...plugin,
+			...wporgPlugin,
+		};
+	}, [ plugin, wporgPlugin ] );
+
+	const isPluginInstalledOnsite = useMemo( () => {
+		if ( requestingPluginsForSites ) {
+			return null;
+		}
+
+		return !! props.sitePlugin;
+	}, [ requestingPluginsForSites, props.sitePlugin ] );
+
+	const existingPlugin = useMemo( () => {
+		if ( isFetching || ! isFetched ) {
+			return 'unknown';
+		}
+		if ( fullPlugin && fullPlugin.fetched ) {
+			return true;
+		}
+
+		// If the plugin has at least one site then we know it exists
+		const pluginSites = Object.values( fullPlugin.sites );
+		if ( pluginSites && pluginSites[ 0 ] ) {
+			return true;
+		}
+
+		if ( requestingPluginsForSites ) {
+			return 'unknown';
+		}
+
+		return false;
+	}, [ isFetching, isFetched, fullPlugin, requestingPluginsForSites ] );
+
+	function getPageTitle() {
```

Comment:
nitpick: let's follow the fun "arrow function" pattern when declaring new functions.
```suggestion
	const getPageTitle = () => {
```